src/Service.Tests/Unittests/MultiSourceQueryExecutionUnitTests.cs

```diff
@@ -0,0 +1,143 @@
+            };
+
+            Dictionary<string, DataSource> dataSourceNameToDataSource = new()
+            {
+                { dataSourceName1, new(DatabaseType.MySQL, "Server =<>;Database=<>;User=xyz;Password=xxx", new())},
+                { dataSourceName2, new(DatabaseType.CosmosDB_NoSQL, "Server =<>;Database=<>;User=xyz;Password=xxx", new()) }
+            };
+
+            Dictionary<string, string> entityNameToDataSourceName = new()
+            {
+                { entityName1, dataSourceName1 },
+                { entityName2, dataSourceName2 }
+            };
+
+            RuntimeConfig mockConfig1 = new(
+               Schema: "",
+               DataSource: new(DatabaseType.MySQL, "Server =<>;Database=<>;User=xyz;Password=xxx", new()),
+               Runtime: new(
+                   Rest: new(),
+                   GraphQL: new(),
+                   Host: new(null, null)
+               ),
+               DefaultDataSourceName: dataSourceName1,
+               DataSourceNameToDataSource: dataSourceNameToDataSource,
+               EntityNameToDataSourceName: entityNameToDataSourceName,
+               Entities: new(entities)
+               );
+
+            string jsonResult = "{\"FirstName\": \"db1\"}";
+
+            JsonDocument document1 = JsonDocument.Parse(jsonResult);
+            Tuple<JsonDocument, IMetadata> mockReturn1 = new(document1, PaginationMetadata.MakeEmptyPaginationMetadata());
+
+            string jsonResult2 = "{\"Name\":\"db2\"}";
+            JsonDocument document2 = JsonDocument.Parse(jsonResult2);
+            Tuple<JsonDocument, IMetadata> mockReturn2 = new(document2, PaginationMetadata.MakeEmptyPaginationMetadata());
+
+            Mock<IQueryEngine> sqlQueryEngine = new();
+            sqlQueryEngine.Setup(x => x.ExecuteAsync(It.IsAny<IMiddlewareContext>(), It.IsAny<IDictionary<string, object>>(), dataSourceName1)).Returns(Task.FromResult(mockReturn1));
+
+            Mock<IQueryEngine> cosmosQueryEngine = new();
+            cosmosQueryEngine.Setup(x => x.ExecuteAsync(It.IsAny<IMiddlewareContext>(), It.IsAny<IDictionary<string, object>>(), dataSourceName2)).Returns(Task.FromResult(mockReturn2));
+
+            Mock<IQueryEngineFactory> queryEngineFactory = new();
+            queryEngineFactory.Setup(x => x.GetQueryEngine(DatabaseType.MySQL)).Returns(sqlQueryEngine.Object);
+            queryEngineFactory.Setup(x => x.GetQueryEngine(DatabaseType.CosmosDB_NoSQL)).Returns(cosmosQueryEngine.Object);
+
+            Mock<IMutationEngineFactory> mutationEngineFactory = new();
+
+            Mock<RuntimeConfigLoader> mockLoader = new(null);
+            mockLoader.Setup(x => x.TryLoadKnownConfig(out mockConfig1, It.IsAny<bool>())).Returns(true);
+
+            RuntimeConfigProvider provider = new(mockLoader.Object);
+
+            string graphQLSchema = File.ReadAllText("MultiSourceTestSchema.gql");
+            ISchemaBuilder schemaBuilder = SchemaBuilder.New().AddDocumentFromString(graphQLSchema)
```

Comment:
can you add clarifying comments about this? is the `testschema.gql` used for comparisons? I don't recall a graphql schema file being used other than for Cosmos.

Reply:
Added more comments. I have created a mock schema that can be used to test accurate routing of multi-source queries by using the schema file. I saw other tests also had schema files but they declared it as string constants in the code. Since this was a large file with multiple entities, i thought it made more sense just to store it on filesystem and use this to unit-test multi-source query execution.